pkg/cmd/roachtest/tests/cdc.go

```diff
@@ -2120,6 +2187,62 @@ func (k kafkaManager) installJRE(ctx context.Context) error {
 	})
 }
 
+// installAzureCli installs azure cli on the kafka node which is necessary for
+// retrieving endpoint connection string for the roachtest azure event hub.
+func (k kafkaManager) installAzureCli(ctx context.Context) error {
+	k.t.Status("installing azure cli")
+	retryOpts := retry.Options{
+		InitialBackoff: 1 * time.Minute,
+		MaxBackoff:     5 * time.Minute,
+	}
+	return retry.WithMaxAttempts(ctx, retryOpts, 3, func() error {
+		return k.c.RunE(ctx, k.nodes, installAzureCliScript)
+	})
+}
+
+// getConnectionString retrieves the Azure Event Hub connection string for the
+// cdc-roachtest event hub set up in the CRL Azure account for roachtest
+// testing.
+func (k kafkaManager) getConnectionString(ctx context.Context) (string, error) {
+	// The necessary credential env vars have been added to TeamCity agents by
+	// dev-inf. Note that running this test on roachprod would not work due to
+	// lacking the required credentials env vars set up.
+	azureClientID := os.Getenv("AZURE_CLIENT_ID")
+	azureClientSecret := os.Getenv("AZURE_CLIENT_SECRET")
+	azureSubscriptionID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+	azureTenantID := os.Getenv("AZURE_TENANT_ID")
+
+	k.t.Status("getting azure event hub connection string")
+	// az login --service-principal -t <Tenant-ID> -u <Client-ID> -p=<Client-secret>
+	cmdStr := fmt.Sprintf("az login --service-principal -t %s -u %s -p=%s", azureTenantID, azureClientID, azureClientSecret)
+	results, err := k.c.RunWithDetailsSingleNode(ctx, k.t.L(), k.nodes, cmdStr)
+	k.t.L().Printf("az login%s\n", results.Stdout)
```

Comment:
I think we should consider using `redact.RedactableString` in order to redact secrets from stdout [1]. While the test environment doesn't have _real_ secrets (i.e., devoid of customer data), each run and its artifacts are _publicly_ accessible. So, in theory a bad actor could infiltrate the test account if all auth. secrets are logged in plaintext.

[1] https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/1824817806/Log+and+error+redactability

Reply:
I don't think the output from az login doesn't contain any secrets. But the output from `az account set` and `az eventhubs namespace` does, so I am intentionally now printing their outputs. I will slack you to confirm if the output from az login contains any secrets (not doing it here to avoid secret leak on github).

Reply:
I think I will just remove this output from our log to make us feel safer. The error message here seems to be sufficient. Just pushed the changes.